src/semigrp.cc

```diff
@@ -42,13 +42,13 @@ using libsemigroups::UNDEFINED;
 using libsemigroups::word_type;
 
 #ifdef SEMIGROUPS_KERNEL_DEBUG
-#define ERROR(obj, message)                               \
-  char buf[128];                                          \
-  strncpy(buf, __func__, sizeof(buf));                    \
-  strncat(buf, ": ", (sizeof(buf) * 2));                  \
-  strncat(buf, message, (sizeof(buf) * strlen(message))); \
-  strncat(buf, " not a %s,", (sizeof(buf) * 9));          \
-  ErrorQuit(buf, (Int) TNAM_OBJ(obj), 0L);
+#define ERROR(obj, message) \
+  static std::string buf;   \
+  buf += __func__;          \
+  buf += ": ";              \
+  buf += message;           \
+  buf += " not a %s,";      \
```

Comment:
Why does the error message end with a trailing comma?

Also, since you are creating buf here anyway, you could also replace this:
```suggestion
  buf += " not a ";      \
  buf += TNAM_OBJ(obj);      \
  buf += ",";      \
```
and then don't pass TNAM_OBJ to ErrorQuit

Reply:
Fair enough, the trailing comma is standard in the Semigroups package error messages, to indicate where the error message ends and the stuff output by GAP begins, if you see what I mean. This might not any longer be necessary/desirable, in the (ever more distant past) error messages in GAP were not very good/accurate and so we used a particular format in the error message strings as a hack to make up for the shortcomings.

Reply:
PS: the reason for this PR is that the travis tests are failing. So far, I've been unable to figure out what is causing these failures, and I also haven't been able to reproduce them locally.